custom_components/nordpool/__init__.py

```diff
@@ -64,23 +92,54 @@ async def _update(self, type_="today", dt=None):
         # when the region is in another timezone
         # as we request data for 3 days anyway.
         # Keeping this for now, but this should be changed.
+        attemps = []
         for currency in self.currency:
             spot = AioPrices(currency, client)
             data = await spot.hourly(end_date=dt)
-            if data:
+            # We only verify the the areas that has the correct currency, example AT is always inf for all other currency then EUR
+            # Now this will fail for any users that has a non local currency for the region they selected.
+            # Thats a problem for another day..
+            regions_to_verify = [k for k, v in _REGIONS.items() if v[0] == currency]
+            data_ok = test_valid_nordpooldata(data, region=regions_to_verify)
+            attemps.append(data_ok)
+            if data_ok is False:
+                np_should_have_released_new_data = stock(dt).replace(
+                    hour=13, minute=RANDOM_MINUTE, second=RANDOM_SECOND
+                )
+
+                if type_ == "tomorrow":
+                    if stock(dt) >= np_should_have_released_new_data:
+                        _LOGGER.info(
+                            "The time is %s, but nordpool havnt released any new data retrying in 5 minutes",
+                            dt,
+                        )
+                        p = partial(self._update, type_, dt)
+                        async_call_later(hass, 60 * 5, p)
+                    else:
+                        _LOGGER.debug("No new data is availble yet")
+
+                else:
+                    _LOGGER.debug("Retrying request for %s", type_)
+                    p = partial(self._update, type_, dt)
+                    async_call_later(hass, 60 * 5, p)
```

Comment:
I'd better create a loop function that would mostly sit in asyncio.sleep and periodically would call an update function. Depending on the update result that would configure next sleep cycle. But that is a code style preference.
Then update function would be much simpler and more readable.

Reply:
Thanks for the suggestion! I'll use backoff for this. It will be something like this. https://replit.com/@Hellowlol/MoccasinSeparateDesignmethod#main.py

Reply:
New dependency? Do not see that homeassistant uses it.
Found this:
https://github.com/home-assistant/core/blob/9f8c0685e35da0ee4ba693395be5d2f77917ea3f/homeassistant/components/apple_tv/__init__.py#L215

Reply:
Yes it is. It dont matter if core using it or not it will be installed if somebody uses this integration